src/Symfony/Component/Mailer/Bridge/AhaSend/Tests/Transport/AhaSendApiTransportTest.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\AhaSend\Tests\Transport;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\JsonMockResponse;
+use Symfony\Component\Mailer\Bridge\AhaSend\Event\AhaSendDeliveryEvent;
+use Symfony\Component\Mailer\Bridge\AhaSend\Transport\AhaSendApiTransport;
+use Symfony\Component\Mailer\Envelope;
+use Symfony\Component\Mailer\Header\TagHeader;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Mime\Part\DataPart;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class AhaSendApiTransportTest extends TestCase
+{
+    /**
+     * @dataProvider getTransportData
+     */
+    public function testToString(AhaSendApiTransport $transport, string $expected)
+    {
+        $this->assertSame($expected, (string) $transport);
+    }
+
+    public static function getTransportData()
+    {
+        return [
+            [
+                new AhaSendApiTransport('KEY'),
+                'ahasend+api://send.ahasend.com',
+            ],
+            [
+                (new AhaSendApiTransport('KEY'))->setHost('example.com'),
+                'ahasend+api://example.com',
+            ],
+            [
+                (new AhaSendApiTransport('KEY'))->setHost('example.com')->setPort(99),
+                'ahasend+api://example.com:99',
+            ],
+        ];
+    }
+
+    public function testSend()
```

Comment:
Please use MockHttpClient (no mock) with a JsnMockResponse (also no mock)

Reply:
I've made some changes (followed what the Postal mailer bridge is doing). Please let me know if it looks good.